### PR TITLE
PC-185: Reimplement the 'submission to journal' feature

### PR DESCRIPTION
**Pull request by marta-**
Reimplemented as an extension of the AccessRightsManager widget, instead of modifying the code of the widget directly.
IMPORTANT: Depends on [PT-3324](https://phenotips.atlassian.net/browse/PT-3324), so it will work only if PhenomeCentral is built on top of a PhenoTips version where either PR phenotips/phenotips#2379 or PR phenotips/phenotips#2380 is merged.
TODOs:
* the list of journals is still hardcoded
* the UI is not translatable.

### CODE REVIEW

**Comment by veronikaslc** [comment]
I think `JournalSubmission.xml` is missing parent

**Comment by marta-** [comment]
Thanks @veronikaslc . Somehow I always forget to add the parent. I made 2 more commits:
* added the parent
* made sure the journal submission option is shown only for patients, not families (@sdumitriu)

**Comment by veronikaslc** [comment]
compiles file, the only thing is - after getting "ClinicalGenetics" document, `$jDoc.isNew()` returns `true`, so journals list is empty and nothing gets generated. Should it be like that?

**Comment by marta-** [comment]
@veronikaslc That is normal on a fresh build with no data, since the journal's account is not part of the code base.

File: ui/src/main/resources/PhenomeCentral/JournalSubmission.xml [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+        #set ($jDoc = $xwiki.getDocument("XWiki.$jId"))
+        #if (!$jDoc.isNew())
+          #if (!$isFirst)
+          ,
+          #end
+          {
+            "id" : "$services.model.serialize($jDoc.documentReference, 'default')",
+            "name" : "$xwiki.getUserName("XWiki.$jId", false)",
+            "type"        : "user",
+            "accessLevel" : "manage"
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenomeCentral/JournalSubmission.xml]: This doesn't work as expected, the journal user only gets `view` access.

File: ui/src/main/resources/PhenomeCentral/JournalSubmission.xml [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        var label = new Element('label').insert(input).insert(j.name);
+        var row = new Element('tr')
+               .insert(new Element('td').insert(label));
+        table.insert(row);
+
+        input.observe('change', function(event) {
+          if (event.target) {
+            if (event.target.checked === false) {
+              // Remove journal collaborator
+              console.log('Removing journal: ' + j.id);
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenomeCentral/JournalSubmission.xml]: `console.log` usage without first checking it exists, probably will break in IE.

File: ui/src/main/resources/PhenomeCentral/JournalSubmission.xml [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+              _this._removeJournal(j.id);
+            } else {
+              console.log('Adding journal: ' + j.id);
```
Comment (sdumitriu) [inline ui/src/main/resources/PhenomeCentral/JournalSubmission.xml]: Another `console` usage.

File: ui/src/main/resources/PhenomeCentral/JournalSubmission.xml (hunk 1 above)
Comment (sdumitriu) [reply ui/src/main/resources/PhenomeCentral/JournalSubmission.xml]: Aha, it's because the expected property is `level`, not `accessLevel`.

**Pull request merged**